### PR TITLE
[profiling] Disable oom monitoring on windows

### PR DESCRIPTION
**Pull request by nsavoire**
### What does this PR do?
OOM monitoring currently does not work on Windows (process aborts before being able to send a last profile). This commit disables it by default on Windows.

### Security 
Datadog employees:
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Comment by szegedi** [comment]
With #3969 I most likely caused a merge conflict for you; sorry!

**Comment by nsavoire** [comment]
> With #3969 I most likely caused a merge conflict for you; sorry!

No worry, I was expecting it !

**Review by szegedi — APPROVED** [review]
LGTM, but I'll have to reapprove after merge conflict resolution

**Review by Qard — APPROVED** [review]

**Pull request merged**
